### PR TITLE
Add method to shutdown and collect results before at_exit 

### PR DESCRIPTION
**Pull request by macumber**
We are using simplecov as part of a testing framework and have need to collect results before `at_exit`.  The new `end_now` method allows the user to get results before `at_exit`.

### CODE REVIEW

**Comment by macumber** [comment]
@colszowka I can add a test for this new method, wanted to check if you were ok with the general premise first though

**Comment by PragTob** [comment]
:wave: 

Hi there,

sorry for the delay I've been rather busy :) What's you use case if I might ask? It's rather specific. To understand if there'd be a better way to do this/accomplish this I need to understand the use case.

File: lib/simplecov.rb [hunk 1]
```diff
@@ -53,6 +53,17 @@ def start(profile = nil, &block)
...
     end
 
+    #
+    # Method added by NREL to shutdown SimpleCov and collect results
+    # this will set running to false which will prevent from running again at_exit
+    def end_now
+      SimpleCov.set_exit_exception
+      exit_status = SimpleCov.exit_status_from_exception
+      SimpleCov.result.format!
+      exit_status = SimpleCov.process_result(SimpleCov.result, exit_status)
```
Comment (PragTob) [inline lib/simplecov.rb]: I think `SimpleCov` should be `self` here so no need to address it directly imo. Also I think the last assignment and return aren't needed :)

File: lib/simplecov.rb [hunk 2]
```diff
@@ -53,6 +53,17 @@ def start(profile = nil, &block)
       Coverage.start
     end
 
+    #
+    # Method added by NREL to shutdown SimpleCov and collect results
```
Comment (PragTob) [inline lib/simplecov.rb]: What is NREL?

**Comment by PragTob** [comment]
@macumber Hi there! :wave: Any plans on picking this up or detailing your use case for this? Would be really helpful! :)

**Comment by deivid-rodriguez** [comment]
Closing this because it's being included in #780. @macumber the linked PR achieves the same thing you were looking for here. Only difference is that the "early exit method" is called `SimpleCov.run_exit_tasks!`, not `SimpleCov.end_now`.

**Pull request closed without merging**